signingscript/src/signingscript/sign.py

```diff
@@ -1512,7 +1536,38 @@ async def rcodesign_notarize(paths, notarization_workdir, creds_path):
             creds_path,
             app_path,
         ]
-        await utils.execute_subprocess(command)
+
+        # Similar implementation to utils.execute_subprocess, but handling some errors:
+        message = 'Running "{}"'.format(" ".join(command))
+        log.info(message)
+        subprocess = await asyncio.create_subprocess_exec(
+            *command,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.STDOUT,
+        )
+        log.info("COMMAND OUTPUT: ")
+        output_lines = []
+        while True:
+            line = await subprocess.stdout.readline()
+            if not line:
+                break
+            line = line.decode("utf-8").rstrip()
+            log.info(line)
+            output_lines.append(line)
+        exitcode = await subprocess.wait()
+        log.info("exitcode {}".format(exitcode))
+        if exitcode > 0:
+            if not output_lines or "Connection reset by peer" not in output_lines[-1]:
+                raise SigningScriptError("Unknown error from notarization service.")
+            log.info("Connection reset by peer, attempting to recover")
+            for line in output_lines:
+                if "submission ID: " in line:
+                    submission_id = line.split(": ")[-1]
+                    break
+            else:
+                raise SigningScriptError("Unable to find notarization submission ID from logs.")
+            await rcodesign_notary_wait(submission_id, creds_path)
```

Comment:
What happens if this commands return an error? It looks like we'll bail immediately, without any retries? It seems like `notary_wait` ought to be called in a loop, or at the very least with multiple retries.

(In fact, maybe `notary-submit` should be retried as well.)